reputation-bar/reputation-bar.lua

```diff
@@ -16,39 +16,62 @@
 ---	See the License for the specific language governing permissions and
 ---	limitations under the License.
 ----------------------------------------------------------------------------------
+---
+
+local Ellyb = Ellyb(...);
 
 ---@type StatusBar
 local StorylineReputationBar = Storyline_NPCFriendshipStatusBar;
 
--- API
-local GetFriendshipReputation = GetFriendshipReputation;
-
-
 local customReputationColors = {
 	[1391545]   = CreateColor(0.227,0.203,0.745, 1), -- Arcane thirst of the Nightfallen
 	["DEFAULT"] = CreateColor(0.709,0.396,0.031, 1),
 }
 local DEFAULT_ICON = [[Interface\Common\friendship-heart]];
 
-hooksecurefunc("NPCFriendshipStatusBar_Update", function(_, factionID --[[ = nil ]])
-	local id, rep, maxRep, name, text, texture, reaction, threshold, nextThreshold = GetFriendshipReputation(factionID);
-	StorylineReputationBar.friendshipFactionID = id;
-	if ( id and id > 0 ) then
-		if ( not nextThreshold ) then
-			threshold, nextThreshold, rep = 0, 1, 1;
+local function OnReputationBarLoad(self)
+	self:SetStatusBarTexture(1, 1, 1, "BORDER", -1);
+	self:GetStatusBarTexture():SetGradient("VERTICAL", Ellyb.Color.CreateFromRGBA(8/255, 93/255, 72/255, 1), Ellyb.Color.CreateFromRGBA(11/255, 136/255, 105/255, 1));
+end
+
+local function OnReputationBarEnter(self)
+	ShowFriendshipReputationTooltip(self.friendshipFactionID, self, "ANCHOR_TOPRIGHT")
```

Comment:
This function was moved to a mixin; `ReputationBarMixin.ShowFriendshipReputationTooltip`.